src/plugins/discover/public/embeddable/actions/view_saved_search_action.ts

```diff
@@ -29,16 +25,19 @@ export const ACTION_VIEW_SAVED_SEARCH = 'ACTION_VIEW_SAVED_SEARCH';
 
 type ViewSavedSearchActionApi = CanAccessViewMode & HasType & PublishesSavedSearch;
 
-const compatibilityCheck = (
-  api: EmbeddableApiContext['embeddable']
-): api is ViewSavedSearchActionApi => {
-  return (
+/** Async type guards aren't supported, so need to make this an async getter instead */
+const getCompatibilityCheck = async (): Promise<
+  (api: EmbeddableApiContext['embeddable']) => api is ViewSavedSearchActionApi
+> => {
+  const { apiCanAccessViewMode, apiHasType, apiIsOfType, getInheritedViewMode } = await import(
+    '@kbn/presentation-publishing'
+  );
```

Comment:
One downside to this approach is that I think our async bundle will now grow with `@kbn/presentation-publishing` since we're importing the entire package here. I wonder if it would work to keep the imports how they were for tree shaking, but break `getCompatibilityCheck` out into a separate file and async import that here instead?

Reply:
Oh!! Genius 🤯 Good call on that one - did this in 5080816909847643940a05f1cbf34c172bdc3cd4 and it reduced the async bundle size that this adds by ~6KB (I think it was around +12KB before).

Reply:
Awesome 🤘 thank you!